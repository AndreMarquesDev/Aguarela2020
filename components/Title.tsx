import React, { FC } from 'react';

interface ITitleProps {
    text: string;
}

const Title: FC<ITitleProps> = props => {
    const {
        text,
    } = props;

    return (
        <>
            <h1 className="champagneBold">
                {text}
            </h1>

            <style jsx>
                {`
                    h1 {
                        font-size: 30rem;
                        text-transform: uppercase;
                        margin-bottom: 30rem;
                    }
                `}
            </style>
        </>
    );
};

export default Title;
