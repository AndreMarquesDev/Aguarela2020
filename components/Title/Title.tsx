import React from 'react';

export interface TitleProps {
    text: string;
    colored?: boolean;
    marginBottom?: number;
}

export const Title = ({ text, colored, marginBottom }: TitleProps): JSX.Element => (
    <>
        <h1 className={colored && 'colored'}>{text}</h1>

        <style jsx>
            {`
                @import './styles/_vars.scss';

                h1 {
                    @include fontL($white, uppercase, bold);
                    text-align: center;
                    margin-bottom: ${marginBottom || 50}rem;
                }

                .colored {
                    color: $pink;
                }
            `}
        </style>
    </>
);
