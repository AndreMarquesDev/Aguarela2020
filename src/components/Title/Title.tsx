import classNames from 'classnames';
import React from 'react';

export type TitleProps = {
    text: string;
    colored?: boolean;
    marginBottom?: number;
};

export const Title = ({ text, colored, marginBottom }: TitleProps): JSX.Element => (
    <>
        <h1 className={classNames('fontL', colored ? 'colored' : '')}>{text}</h1>

        <style jsx>
            {`
                h1 {
                    color: var(--white);
                    text-transform: uppercase;
                    font-weight: bold;
                    text-align: center;
                    margin-bottom: ${marginBottom || 50}rem;
                }

                .colored {
                    color: var(--pink);
                }
            `}
        </style>
    </>
);
