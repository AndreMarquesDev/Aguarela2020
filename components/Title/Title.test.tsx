import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Title, TitleProps } from './Title';

const baseProps: TitleProps = {
    text: 'dummy text',
    colored: false,
    marginBottom: null,
};

describe('<Title />', () => {
    test('renders properly', () => {
        render(<Title {...baseProps} />);

        const h1Element = screen.getByText(baseProps.text);

        expect(h1Element).toBeInTheDocument();
        expect(h1Element).toMatchSnapshot();
    });

    test('renders properly with the colored class', () => {
        render(<Title {...baseProps} colored />);

        const h1Element = screen.getByText(baseProps.text);

        expect(h1Element).toBeInTheDocument();
        expect(h1Element).toHaveClass('colored');
        expect(h1Element).toMatchSnapshot();
    });

    test('renders properly with a custom margin-bottom', () => {
        render(<Title {...baseProps} marginBottom={100} />);

        const h1Element = screen.getByText(baseProps.text);

        expect(h1Element).toBeInTheDocument();
        expect(h1Element).toMatchSnapshot();
    });
});
