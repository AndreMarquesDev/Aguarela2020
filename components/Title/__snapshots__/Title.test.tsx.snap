// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders properly 1`] = `
<h1
  class="jsx-3155756322 "
>
  dummy text
</h1>
`;

exports[`<Title /> renders properly with a custom margin-bottom 1`] = `
<h1
  class="jsx-3426305334 "
>
  dummy text
</h1>
`;

exports[`<Title /> renders properly with the colored class 1`] = `
<h1
  class="jsx-3155756322 colored"
>
  dummy text
</h1>
`;
